src/toil/wdl/wdltoil.py

```diff
@@ -508,44 +1030,162 @@ def run(self, file_store: AbstractFileStore) -> WDLBindings:
             # TODO: MiniWDL can parallelize the fetch
             bindings = devirtualize_files(bindings, standard_library)
 
-            # Work out the command string, and unwrap it
-            command_string: str = evaluate_named_expression(self._task, "command", WDL.Type.String(), self._task.command, bindings, standard_library).coerce(WDL.Type.String()).value
+            # Make the container object
+            # TODO: What is this?
+            run_id = str(uuid.uuid4())
+            # Directory on the host where the conteiner is allowed to put files.
+            host_dir = os.path.abspath('.')
+            # Container working directory is guaranteed (?) to be at "work" inside there
+            workdir_in_container = os.path.join(host_dir, "work")
+            task_container = TaskContainerImplementation(miniwdl_config, run_id, host_dir)
+
+            # Until miniwdl 1.10+ is released, we need to replace singularity
+            # run with singularity exec to handle overriding container
+            # entrypoints (see
+            # <https://github.com/chanzuckerberg/miniwdl/issues/628>). Also, we
+            # might need to send GPUs and the current miniwdl deosn't do that
+            # for Singularity. And we might need to *not* try and use
+            # --fakeroot if we lack sub-UIDs. So we sneakily monkey patch it
+            # here.
+            original_run_invocation = task_container._run_invocation
+            def patched_run_invocation(*args: Any, **kwargs: Any) -> List[str]:
+                """
+                Invoke the original _run_invocation to get a base Singularity
+                command line, and then adjust the result to use "exec" and to
+                pass GPUs if needed.
+                """
+                # We are written against https://github.com/chanzuckerberg/miniwdl/blob/d8481a01000118eda76cbbb30df9fa5df493b86f/WDL/runtime/backend/singularity.py#L75
+                command_line: List[str] = original_run_invocation(*args, **kwargs)
+
+                logger.debug('MiniWDL wants to run command line: %s', command_line)
+
+                # "run" can be at index 1 or 2 depending on if we have a --verbose. We also might already have just "exec" there.
+                subcommand_index = 2 if command_line[1] == "--verbose" else 1
+                if command_line[subcommand_index] == "run":
+                    command_line[subcommand_index] = "exec"
```

Comment:
The fix might have been included in the `v1.9.1` release yesterday? 
https://github.com/chanzuckerberg/miniwdl/releases/tag/v1.9.1

Reply:
Yeah I can cut this piece and require the new version.